github-actions/start-release/tests/test_start_release.py

```diff
@@ -182,11 +200,23 @@ def test_start_release_happy_path(session,
 
     start_release(session)
 
-    assert session.get.call_count == 5
-    assert session.get.call_args_list[0] == call('contents?ref=next')
-    assert session.get.call_args_list[1] == call('contents/{}.json?ref=next'.format(APP_NAME))
-    assert session.get.call_args_list[2] == call('contents/{}.json?ref=main'.format(APP_NAME))
-    assert session.get.call_args_list[3] == call('git/ref/heads/next')
+    if isinstance(app_json_main, dict):
+        assert session.get.call_count == 7
+        assert session.get.call_args_list[0] == call('contents?ref=next')
+        assert session.get.call_args_list[1] == call('contents/{}.json?ref=next'.format(APP_NAME))
+        assert session.get.call_args_list[2] == call('git/blobs/{}?ref=next'.format(sha_field_next))
+        assert session.get.call_args_list[3] == call('contents/{}.json?ref=main'.format(APP_NAME))
+        assert session.get.call_args_list[4] == call('git/blobs/{}?ref=main'.format(sha_field_main))
+        assert session.get.call_args_list[5] == call('git/ref/heads/next')
+        assert session.get.call_args_list[6] == call('pulls')
+    else:
+        assert session.get.call_count == 6
+        assert session.get.call_args_list[0] == call('contents?ref=next')
+        assert session.get.call_args_list[1] == call('contents/{}.json?ref=next'.format(APP_NAME))
+        assert session.get.call_args_list[2] == call('git/blobs/{}?ref=next'.format(sha_field_next))
+        assert session.get.call_args_list[3] == call('contents/{}.json?ref=main'.format(APP_NAME))
```

Comment:
why does the `blobs` call not happen in this case? is it because the contents call fails for `main`?

Reply:
Yes, when main json is empty, the content call will be failed with HTTPError (not found), then, the gitblob call will not be happened.